src/mod/l4_thruster.c

```diff
@@ -541,27 +741,1590 @@ void GeneralReadRequest(int argc, char *argv[]){
+				THR_HARDCODED_SEQUENCES[procedure_id].substage_index = 0;
+
+				//*******assume that argv[5] is custom print message !!!!! WARNING I AM NOT SURE THAT THIS IS GOOD IDEA
+				len = strlen(argv[5]);
+				deb_print_pure_debug((uint8_t *)argv[5], len);
+				//// WARNING THIS BLOCK MIGHT BE NO GOOD *********
+			}
+
+
+
+
+		break;
+	case 1:
+		sprintf(print_str, "\n Sending READ [%d] Request %d\n",register_index, now_timestamp );
+		len = strlen(print_str);
+		deb_print_pure_debug((uint8_t *)print_str, len);
+
+		temp_argv[0]= "6";
+		//temp_argv[1]= "97"; // Reservoir Temperature
+
+		sprintf(argument1, "%d",register_index);
+		temp_argv[1]= argument1;
+
+		GeneralReadRequest(2, temp_argv);
+
+		THR_HARDCODED_SEQUENCES[procedure_id].substage_index ++;
+		THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_substage = (uint32_t)timGetSystime();
+
+		break;
+	case 2:
+
+		if ( (now_timestamp - THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_substage) < wait_for_read_request_proccess ){ // Wait to proccess READ request
+						// do nothing
+					}
+		else{
+			//THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_substage = (uint32_t)timGetSystime(); //save execution finish time of wait substage
+
+			sprintf(print_str, "\nMonitored [%d] Value = %.6f \n", register_index,REGISTER_DATA[register_index]);
+			len = strlen(print_str);
+			deb_print_pure_debug((uint8_t *)print_str, len);
+			THR_HARDCODED_SEQUENCES[procedure_id].substage_index = 0;
+
+
+			}
+
+
+		break;
+
+	}
+
+
+
+
+
+
+}
+
+
+void thr_void(int argc, char *argv[]){
+	LAST_STARTED_MODULE = 1109;
+
+	/*
+	 * Incorporates with sequence execution
+	 *
+	 * void function that does nothing
+	 */
+
+	uint16_t procedure_id = atoi(argv[0]); // procedure_id is always fist index of argument array
+	THR_HARDCODED_SEQUENCES[procedure_id].execution_index ++;
+
+}
+
+void thr_value_ramp(int argc, char *argv[]){
+	/*
+	 * According to enpulsion defination RAMP Frequency should be 1 Hz, Duration of ramp is 30s
+	 * Out function allows to specify different dt(frequency) and ramp duration.
+	 * When hardcoding sequences - lets stick with enpulsion ramp requirments
+	 */
+
+	//int substage_index =0;
+
+	uint16_t procedure_id = atoi(argv[0]); // procedure_id is always fist index of argument array
+	uint8_t register_index = atoi(argv[1]); // first argument is register index at which SET ramp would be implemented
+	double goal = atof((const char*)argv[2]); // goal to which value should be set
+	uint32_t ramp_duration = atoi(argv[3]); // time through which value should be changed from initial to goal
+	uint32_t ramp_dt = atoi(argv[4]); // wait between SET / Converted to (ms)
+	double initial_value;
+	uint32_t now_timestamp;
+
+	char print_str[200];
+	int len;
+
+	double ramp_iterations = ramp_duration/ ramp_dt; // WARING - HARDCODE IN A WAY THAT THIS IS ALWAYS INT
+	double value_step;
+
+	char *temp_argv[3];
+	//char argument1[1];
+	char argument2[50];
```

Comment:
Yes, this stupidity again. 

but no other way if needed to make it work with argv ....